intelmq/bots/parsers/netlab_360/parser.py

```diff
@@ -21,18 +22,28 @@ def parse_line(self, line, report):
 
             if report['feed.url'] in Netlab360ParserBot.DGA_FEED:
                 event.add('source.fqdn', value[1])
-                event.add('time.source', value[3] + ' UTC')
+                # DGA Feed format is
+                # DGA family, Domian, Start and end of valid time(UTC)
+
+                event.add('time.source', value[2] + ' UTC')
                 event.add('classification.type', 'c&c')
                 event.add('event_description.url', 'http://data.netlab.360.com/dga')
 
             elif report['feed.url'] in Netlab360ParserBot.MAGNITUDE_FEED:
                 event.add('time.source', DateTime.from_timestamp(int(value[1])))
                 event.add('source.ip', value[2])
-                event.add('source.fqdn', value[3])
+                # ignore ips as fqdns
```

Comment:
What happens when a non-fqdn and non IP address strings comes in here? 
Like for example the string "|||" . That's neither an IP nor a domain.
IMHO a parser should carefully ensure that the input is correct. If not, it must not pass on invalid input.